packages/terra-navigation-side-menu/src/NavigationSideMenu.jsx

```diff
@@ -0,0 +1,203 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import 'terra-base/lib/baseStyles';
+import List from 'terra-list';
+import ActionHeader from 'terra-clinical-action-header';
+import ContentContainer from 'terra-content-container';
+import MenuItem from './_MenuItem';
+
+const KEYCODES = {
+  ENTER: 13,
+};
+
+const propTypes = {
+  /**
+   * An array of configuration for each menu item.
+   */
+  menuItems: PropTypes.arrayOf(PropTypes.shape({
+    /**
+     * Keys of menu items
+     */
+    childKeys: PropTypes.arrayOf(PropTypes.string),
+    /**
+     * Used to match visual style of a menuItem with children, but when the onChange maps to the sub route of another menu.
+     */
+    hasSubMenu: PropTypes.bool,
+    /**
+     * Whether or not the menu is the primary navigation links in small form factor.
+     */
+    isRootMenu: PropTypes.bool,
+    /**
+     * ID to be applied to the menu item div.
+     */
+    id: PropTypes.string,
+    /**
+     * Optional meta data to be returned along with the item key within the onChange.
+     */
+    metaData: PropTypes.object,
+    /**
+     * Unique identifier that will be returned in the onChange callback when an endpoint is reached.
+     */
+    key: PropTypes.string.isRequired,
+    /**
+     * Text for the menu row and header title when selected.
+     */
+    text: PropTypes.string.isRequired,
+  })),
+  /**
+   * Callback function when a menu endpoint is reached.
+   * returns (event, { selectedMenuKey: String, selectedChildKey: String, metaData: Object})
+   */
+  onChange: PropTypes.func.isRequired,
+  /**
+   * Delegate prop showParent function that is provided by the terra-navigation-layout.
+   */
+  routingStackBack: PropTypes.func,
+  /**
+   * Key of the currently selected child item on the selected menu page.
+   * This is used when traveling back up the menu stack or when the child is an end point.
+   */
+  selectedChildKey: PropTypes.string,
+  /**
+   * Key of the currently selected menu page.
+   */
+  selectedMenuKey: PropTypes.string.isRequired,
+};
+
+const defaultProps = {
+  menuItems: [],
+};
+
+const processMenuItems = (menuItems) => {
+  const items = {};
+  const parents = {};
+  menuItems.forEach((item) => {
+    items[item.key] = {
+      id: item.id,
+      text: item.text,
+      childKeys: item.childKeys,
+      metaData: item.metaData,
+      hasSubMenu: item.hasSubMenu,
+      isRootMenu: item.isRootMenu,
+    };
+    if (item.childKeys) {
+      item.childKeys.forEach((key) => {
+        parents[key] = item.key;
```

Comment:
To clarify, is this saving the parents or the children to the parent object with the child key?

Reply:
This is mapping the childKeys to a value of the parentKey.